src/callbacks_step/save_restart_dg.jl

```diff
@@ -157,7 +157,9 @@ function load_restart_file(mesh::Union{ParallelTreeMesh, ParallelP4estMesh}, equ
   if !mpi_isroot()
     # Receive nodal data from root
     for v in eachvariable(equations)
-      MPI.Scatterv!(nothing, @view(u[v, .., :]), mpi_root(), mpi_comm())
+      if length(u[v, .., :]) > 0
+        MPI.Scatterv!(nothing, @view(u[v, .., :]), mpi_root(), mpi_comm())
+      end
```

Comment:
Are you sure this approach works? Because it looks to me like you are now only call Scatterv from ranks with data, but then the call will deadlook because this is a collective *blocking* operation and you're using `mpi_comm()` (which defaults to `MPI_COMM_WORLD`), and thus all other ranks will be waiting on that one rank that never calls Scatterv...

Reply:
I was also thinking that this might cause a deadlock, but I tested it locally and it ran through.

Reply:
With the `MPI.Barrier` no rank reaches the "reached: " `println` and all but the empty rank (and of course the root rank) reach the "Scatterv" `println`.  Without the `MPI.Barrier` all ranks reach the "reached: " `println`.

Reply:
Then I don’t know why the test would work at all, but it definitely is currently broken (both the test and you’re fix, I’m afraid)

Reply:
Do you have any other idea how to fix the problem?

Reply:
Instead of not communicating from empty ranks, you need to create a (maybe) an auxiliary data structure that is empty and then make sure that the empty ranks set the number of elements they contribute to the communication to zero.

Reply:
Is this ([fc621db](https://github.com/trixi-framework/Trixi.jl/pull/1366/commits/fc621dbf0720b5f58fc10831d268ddbc77b14d27)) what you meant or what exactly do you mean by

> [...] make sure that the empty ranks set the number of elements they contribute to the communication to zero

?

Reply:
At least, this also runs through.

Reply:
Do you think the (potential) problem is fixed now, @sloede?